Update workshop to v1 of SDK (#330)
KILTprotocol/docs: sdk/02_tutorial/07_verification.md

import Verify from '!!raw-loader!@site/code_examples/sdk_examples/src/workshop/verify.ts';
import CreatePresentation from '!!raw-loader!@site/code_examples/sdk_examples/src/workshop/holder/createPresentation.ts';

<!-- Verficiation criteria, expect it  to be tied to Verifier DID, tied to the verficationCriteria, based on four options -->

In this section, you play the role of a <span className="label-role verifier">Verifier</span> that does the following:

:::info

The Verifier also needs an account and DID, but the tutorial omits these steps.

:::

1. Take a `VerifiablePresentation` object supplied by a <span className="label-role holder">Holder</span>
2. Verify that its data is correct

> 3. Verify that the presentation is valid, i.e., its hash exists on-chain and the presentation hasn't been revoked
> 4. Verify that the <span className="label-role holder">Holder</span> sending the `Credential` owns it

Reviewer: ```suggestion
3. Verify that the <span className="label-role holder">Holder</span> of the `Credential` has authorised and consented to its use in the current context by checking the presentation's signature and attributes.  
4. Verify the authenticity and validity of the credential by checking its on-chain proof created by the Issuer and ensuring it hasn't been revoked since
```

:::info VerifiablePresentation object

The <span className="label-role holder">Holder</span> uses a Credential to create the `VerifiablePresentation` object.

> Unlike the credential, a `VerifiablePresentation` can hide some attributes that aren't required by the <span className="label-role verifier">Verifier</span> and can contain a holder-signed challenge.

Reviewer: a Verifiable Presentation actually has an array of credentials, and it's in these credentials that some properties are hidden. so this is not entirely accurate and might be confusing

Author: @rflechtner So is the only change here object > array? Or something else?

Reviewer: The main functional change is that one, yes; in the old Kilt a presentation was simply a credential that was signed by the holder, plus presentation-related data such as a challenge. The signature and challenge and all other presentation-related data is now held by the Verifiable Presentation object, which can have one or more credentials nested within. So while you had one presentation per credential prior, you can now present multiple at the same time using only one signature.

The data model of the presentation has changed significantly though; we now use the data model described in https://www.w3.org/TR/vc-data-model/#presentations-0

A `VerifiablePresentation` also contains a proof that the <span className="label-role holder">Holder</span> owns the credential.

:::

## Verify presentation

The KILT SDK provides the `verifyPresentation` method from the `Verifier` class to verify a credential presentation. It takes the following parameters:

- `presentation`: The `VerifiableCredential` to verify.
- `verficationCriteria`: An object of criteria that need to pass to verify the presentation. These can be any combination of criteria.
<!-- TODO: Find out more link -->

> The method returns a boolean value indicating whether the presentation is valid.

Reviewer: it's an object, and one of this object's properties (`verified`) has a boolean value indicating whether the presentation and credential are valid

1. Take a `VerifiablePresentation` object supplied by a <span className="label-role holder">Holder</span>
2. Verify that its data is correct
3. Verify that the <span className="label-role holder">Holder</span> of the `Credential` has authorised and consented to its use in the current context by checking the presentation's signature and attributes.  

> 4. Verify the authenticity and validity of the credential by checking its on-chain proof created by the Issuer and ensuring it hasn't been revoked since

Reviewer: ```suggestion
4. Verify the authenticity and validity of the credential by checking its on-chain proof created by the Issuer and ensuring it hasn't been revoked since.
```

:::info A VerifiablePresentation object

> The <span className="label-role holder">Holder</span> uses a Credential to create an array of `VerifiablePresentation` objects as .

Reviewer: as ?

Author: ```suggestion
The <span className="label-role holder">Holder</span> uses a Credential to create an array of `VerifiablePresentation` objects.
```
